src/mame/snk/hng64_sprite.ipp

```diff
@@ -40,194 +40,424 @@
-				if (source[2]&0x00000100) // inline chain mode
-					source += 8 * (1 + (source[2]&0x0000000f)) * (1 + ((source[2]&0x000000f0)>>4));
-				else
-					source += 8;
+				uint8_t cb = 0;
+				for (s32 curx = 0; curx < numblocks; curx++)
+				{
+					uint16_t srcpix;
+					PIX_CHECKERBOARD
+					PIXEL_OP_REBASE_TRANSPEN_REV(destptr[0], destzptr[0], srcpix );
+					cursrcx += dx;
+					PIX_CHECKERBOARD
+					PIXEL_OP_REBASE_TRANSPEN_REV(destptr[1], destzptr[1], srcpix);
+					cursrcx += dx;
+					PIX_CHECKERBOARD
+					PIXEL_OP_REBASE_TRANSPEN_REV(destptr[2], destzptr[2], srcpix);
+					cursrcx += dx;
+					PIX_CHECKERBOARD
+					PIXEL_OP_REBASE_TRANSPEN_REV(destptr[3], destzptr[3], srcpix);
+					cursrcx += dx;
+					destptr += 4;
+					destzptr += 4;
+
+				}
+
+				// iterate over leftover pixels
+				for (s32 curx = 0; curx < leftovers; curx++)
+				{
+					uint16_t srcpix;
+					PIX_CHECKERBOARD
+					PIXEL_OP_REBASE_TRANSPEN_REV(destptr[0], destzptr[0], srcpix);
+					cursrcx += dx;
+					destptr++;
+					destzptr++;
+				}
 			}
 			else
-				source += 8;
+			{
+				uint8_t cb = 0;
+				for (s32 curx = 0; curx < numblocks; curx++)
+				{
+					uint16_t srcpix;
+					PIX_CHECKERBOARD
+					PIXEL_OP_REBASE_TRANSPEN(destptr[0], destzptr[0], srcpix);
+					cursrcx += dx;
+					PIX_CHECKERBOARD
+					PIXEL_OP_REBASE_TRANSPEN(destptr[1], destzptr[1], srcpix);
+					cursrcx += dx;
+					PIX_CHECKERBOARD
+					PIXEL_OP_REBASE_TRANSPEN(destptr[2], destzptr[2], srcpix);
+					cursrcx += dx;
+					PIX_CHECKERBOARD
+					PIXEL_OP_REBASE_TRANSPEN(destptr[3], destzptr[3], srcpix);
+					cursrcx += dx;
+					destptr += 4;
+					destzptr += 4;
+
+				}
+
+				// iterate over leftover pixels
+				for (s32 curx = 0; curx < leftovers; curx++)
+				{
+					uint16_t srcpix;
+					PIX_CHECKERBOARD
+					PIXEL_OP_REBASE_TRANSPEN(destptr[0], destzptr[0], srcpix);
+					cursrcx += dx;
+					destptr++;
+					destzptr++;
+				}
+			}
 		}
+	} while (0);
+	g_profiler.stop();
+}
+
+void hng64_state::zoom_transpen(bitmap_ind16 &dest, bitmap_ind16 &destz, const rectangle &cliprect,
+		gfx_element *gfx, u32 code, u32 color, int flipx, int flipy, s32 destx, s32 desty,
+		s32 dx, s32 dy, u32 dstwidth, u32 dstheight, u32 trans_pen, u32 zval, bool zrev, bool blend, bool checkerboard)
+{
+	// use pen usage to optimize
+	code %= gfx->elements();
+	if (gfx->has_pen_usage())
+	{
+		// fully transparent; do nothing
+		u32 usage = gfx->pen_usage(code);
+		if ((usage & ~(1 << trans_pen)) == 0)
+			return;
 	}
 
+	// render
+	color = gfx->colorbase() + gfx->granularity() * (color % gfx->colors());
+
+	if (blend)
+		color |= 0x8000;
+
+	drawgfxzoom_core(dest, destz, cliprect, gfx, code, flipx, flipy, destx, desty, dx, dy, dstwidth, dstheight, trans_pen, color, zval, zrev, checkerboard);
+}
+
+
+
+void hng64_state::draw_sprites_buffer(screen_device& screen, const rectangle& cliprect)
+{
+	m_sprite_bitmap.fill(0x0000, cliprect);
+	gfx_element* gfx;
```

Comment:
Please declare `gfx` at the point before it’s assigned.  The value isn’t valid across iterations of the outer loop, it should always be assigned.